src/main/java/com/unzer/payment/BasketItem.java

```diff
@@ -20,145 +20,287 @@
+	 * If not provided, an ID is automatically generated by Unzer.
+	 */
 	public String getBasketItemReferenceId() {
 		return basketItemReferenceId;
 	}
 
+	/**
+	 * Unique basket item reference ID (within the basket).
+	 * If not provided, an ID is automatically generated by Unzer.
+	 */
 	public BasketItem setBasketItemReferenceId(String basketItemReferenceId) {
 		this.basketItemReferenceId = basketItemReferenceId;
 		return this;
 	}
 
+	/**
+	 * Quantity of the basket item
+	 */
 	public Integer getQuantity() {
 		return quantity;
 	}
 
+	/**
+	 * Quantity of the basket item
+	 */
 	public BasketItem setQuantity(Integer quantity) {
 		this.quantity = quantity;
 		return this;
 	}
 
+	/**
+	 * Amount per unit including VAT.
+	 * <p>
+	 * Note: Discount is not included. Don’t include discounts here!
+	 */
+	public BigDecimal getAmountPerUnitGross() {
+		return amountPerUnitGross;
+	}
+
+	/**
+	 * Amount per unit including VAT.
+	 * <p>
+	 * Note: Discount is not included. Don’t include discounts here!
+	 */
+	public BasketItem setAmountPerUnitGross(BigDecimal amountPerUnitGross) {
+		this.amountPerUnitGross = amountPerUnitGross;
+		return this;
+	}
+
+	/**
+	 * The VAT value for the basket item in percentage
+	 */
 	public Integer getVat() {
 		return vat;
 	}
 
+	/**
+	 * The VAT value for the basket item in percentage
+	 */
 	public BasketItem setVat(Integer vat) {
 		this.vat = vat;
 		return this;
 	}
 
+	/**
+	 * The type of item in the basket. The valid values are goods, voucher, and shipment.
+	 */
+	public Type getType() {
+		return type;
+	}
+
+	/**
+	 * The type of item in the basket. The valid values are goods, voucher, and shipment.
+	 */
+	public BasketItem setType(Type type) {
+		this.type = type;
+		return this;
+	}
+
+	/**
+	 * @deprecated Use enum Type instead of plain string
+	 */
+	@Deprecated
+	public BasketItem setType(String type) {
+		this.type = Type.valueOf(type.toUpperCase());
+		return this;
+	}
+
+	/**
+	 * The measuring unit for the basket item, e.g. pieces/Pc., kilogram/kg
+	 */
+	public String getUnit() {
+		return unit;
+	}
+
+	/**
+	 * The measuring unit for the basket item, e.g. pieces/Pc., kilogram/kg
+	 */
+	public BasketItem setUnit(String unit) {
+		this.unit = unit;
+		return this;
+	}
+
+
+	/**
+	 * Discount (incl. VAT) granted on the basketItems.amountGross.
+	 * Positive amount is expected.
+	 */
+	public BigDecimal getAmountDiscountPerUnitGross() {
+		return amountDiscountPerUnitGross;
+	}
+
+	/**
+	 * Discount (incl. VAT) granted on the basketItems.amountGross.
+	 * Positive amount is expected.
+	 */
+	public BasketItem setAmountDiscountPerUnitGross(BigDecimal amountDiscountPerUnitGross) {
+		this.amountDiscountPerUnitGross = amountDiscountPerUnitGross;
+		return this;
+	}
+
+	public enum Type {
+		@SerializedName("")
+		EMPTY,
+
+		@SerializedName("goods")
+		GOODS,
+
+		@SerializedName("voucher")
+		VOUCHER,
+
+		@SerializedName("shipment")
+		SHIPMENT
+	}
```

Comment:
"digital" is missing.
Please do not just apply the documentation here.
There are some fields not documented that are still relevant and there are still some errors in it.